modules/tasks/tasks/src/main/resources/webapp/js/directives.js

```diff
@@ -537,719 +508,255 @@
-
-                    elementManipulation = elementManipulation.replace(regex, manipulation);
-                    manipulateElement.attr("manipulate", elementManipulation);
-                    manipulateElement.trigger('manipulateChanged');
+            restrict : 'EA',
+            require: '^manipulationSorter',
+            transclude: true,
+            replace: true,
+            templateUrl: '../tasks/partials/manipulation.html',
+            scope: {
+                argument: '=?'
+            },
+            link : function (scope, element, attrs, manipulationSorter) {
+                var attributeFieldTemplate, manipulationSettings = {};
+                ManageTaskUtils.MANIPULATION_SETTINGS.forEach(function(manipulation){
+                    if(attrs.type === manipulation.name){
+                        manipulationSettings = manipulation;
+                    }
                 });
-            }
-        };
-    });
 
-    directives.directive('substringUpdate', function () {
-        return {
-            restrict : 'A',
-            require: '?ngModel',
-            link : function (scope, el, attrs) {
-                el.on("focusout focusin keyup", function (event) {
-                    event.stopPropagation();
-                    var manipulateElement = $("[ismanipulate=true]"),
-                        manipulation = "substring(" + $("#substringSeparator").val() + ")",
-                        elementManipulation = manipulateElement.attr("manipulate"),
-                        regex = new RegExp("substring\\(.*?\\)", "g");
-
-                    elementManipulation = elementManipulation.replace(regex, manipulation);
-                    manipulateElement.attr("manipulate", elementManipulation);
-                    manipulateElement.trigger('manipulateChanged');
-                });
-            }
-        };
-    });
+                scope.msg = scope.$parent.$parent.$parent.msg;
+                scope.type = attrs.type;
+                if(attrs.active){
+                    scope.active = true;
+                }
 
-    directives.directive('dateUpdate', function () {
-        return {
-            restrict : 'A',
-            require: '?ngModel',
-            link : function (scope, el, attrs) {
-                el.on("focusout focusin keyup", function (event) {
-                    event.stopPropagation();
-                    var manipulateElement = $("[ismanipulate=true]"),
-                        manipulation = "dateTime(" + $("#dateFormat").val() + ")",
-                        elementManipulation = manipulateElement.attr("manipulate"),
-                        regex = new RegExp("dateTime\\(.*?\\)", "g");
-
-                    elementManipulation = elementManipulation.replace(regex, manipulation);
-                    manipulateElement.attr("manipulate", elementManipulation);
-                    manipulateElement.trigger('manipulateChanged');
-                });
+                if(attrs.active) {
```

Comment:
This if condition is exactly the same as the previous condition just few lines above. Can those ifs be merged into one?